### PR TITLE
Linkify engine error codes and some refactor.

### LINKED ISSUE #139: Error codes in logs should be linked to documentation
This isn't so important for [tesSUCCESS], but for e.g. [telINSUF_FEE_P], the user will probably be interested in some description of how co estimate a sufficient fee...

fee

### LINKED ISSUE #170: Error codes in debug stream should be linked to documentation
As @RichardAH put it (lightly edited):
Log codes are the numbers that presently appear after some HookSet trace lines, e.g. HookSet(1)[...
Each message type has a unique code so the IDE can link it to some documentation page, to help the user understand what they are doing wrong.

This is related to #139 (it isn't a duplicate mainly because it concerns a different log).

### PR DESCRIPTION
**Pull request by muzam1l**
Fixes #139 
Fixes #170 

~~Engine result codes are all linked to https://xrpl.org/transaction-results.html for now. Comment if we should point each error code separately!~~

### CODE REVIEW

**Comment by vbar** [comment]
> Comment if we should point each error code separately!

I think not every code will have a separate page, but we definitely can have different links for different error prefixes - e.g. `telINSUF_FEE_P` should link to https://xrpl.org/tel-codes.html , not just https://xrpl.org/transaction-results.html ...

**Review by vbar — COMMENTED** [review]
OK, that does fix #139 , but I don't see how it fixes #170 , which is about codes in the debug stream, e.g. when deploying a hook:
    Debug stream opened for account Alice
    2022-Aug-10 08:37:11.621411230 UTC OpenLedger:TRC HookSet(50)[Alice-  7CA32EC8D9273254D8FA4D429F4BA51C8AEB91E24490742413855708E5C1289F]: GuardCheck Total worse-case execution count: 32
    2022-Aug-10 08:37:11.621454238 UTC OpenLedger:TRC HookSet(77)[Alice-7CA32EC8D9273254D8FA4D429F4BA51C8AEB91E24490742413855708E5C1289F]: Trying to wasm instantiate proposed hook size = 257
    2022-Aug-10 08:37:11.662389400 UTC OpenLedger:TRC HookSet(50)[Alice-7CA32EC8D9273254D8FA4D429F4BA51C8AEB91E24490742413855708E5C1289F]: GuardCheck Total worse-case execution count: 32
    ...

These are still not linked.

**Comment by muzam1l** [comment]
Any documentation on those?

**Comment by vbar** [comment]
> Any documentation on those?

Well, in https://github.com/XRPL-Labs/xrpld-hooks/blob/develop/src/ripple/app/hook/Enum.h (enum hook_log_code)... I don't think they have any HTML documentation, and the comments aren't that extensive anyway - maybe they could be displayed in some popup?

**Review by valstu — COMMENTED** [review]
We actually have Tooltip component that @vbar suggested, Should we add question mark -icon or something next to code on debug stream which would open the tooltip?

**Comment by vbar** [comment]
> We actually have Tooltip component that @vbar suggested, Should we add question mark -icon or something next to code on debug stream which would open the tooltip?

I'd just replace the number by its symbolic name and use that to open the tooltip...

**Comment by muzam1l** [comment]
I'd just replace the number by its symbolic name and use that to open the tooltip...

> What content would the tooltip contain?

**Comment by valstu** [comment]
> What content should the tooltip contain then?

I assume, if you get for example `HookSet(77)` you would show the comment of this line https://github.com/XRPL-Labs/xrpld-hooks/blob/develop/src/ripple/app/hook/Enum.h#L125 on the tooltip? @vbar probably knows better

**Comment by vbar** [comment]
> > What content should the tooltip contain then?
> 
> I assume, if you get for example `HookSet(77)` you would show the comment of this line https://github.com/XRPL-Labs/xrpld-hooks/blob/develop/src/ripple/app/hook/Enum.h#L125 (`Informational: first attempt to load wasm into wasm runtime `) on the tooltip?

yes, exactly

**Review by vbar — COMMENTED** [review]
Shouldn't the tooltips in development log and debug stream have the same style? I think the development log tooltip is too large - especially since it isn't saying anything new... :-)

**Comment by muzam1l** [comment]
> tooltips in development log

**Comment by vbar** [comment]
> > tooltips in development log
> 
> Which tooltips are we talking about specifically here 🥹?

Well, when I run a transaction in the test tab and hover over the [tesSUCCESS] link in the development log (which admittedly should be called test log or something :-) ), I see a tooltip (both in Firefox and Chrome) just repeating tesSUCCESS:
tooltip

It isn't styled much, and ideally I think it shouldn't be there at all...

**Comment by muzam1l** [comment]
> It isn't styled much, and ideally I think it shouldn't be there at all...

**Comment by vbar** [comment]
> > It isn't styled much, and ideally I think it shouldn't be there at all...
> 
> Oh that is an `anchor` tag's `title`, just there for accessibility, It ought to be present on every link!

Well, mostly it isn't - the popup help doesn't show it, and neither do links in README... Actually I can't find any links w/ title in the IDE, and I don't think it's all that accessible (nor that it could be)...

**Comment by muzam1l** [comment]
> Actually I can't find any links w/ title in the IDE

They all should have, but for now it is easier to remove title of one link than adding to all others!

File: content/hook-set-codes.json [hunk 1]
```diff
@@ -0,0 +1,434 @@
...
+  {
+    "code": 5,
+    "identifier": "BLOCK_ILLEGAL",
+    "description": " a block end instruction moves execution below depth 0 {{}}`}` <= like this"
+  },
+  {
+    "code": 5,
+    "identifier": "BLOCK_ILLEGAL",
+    "description": " a block end instruction moves execution below depth 0 {{}}`}` <= like this"
+  },
```
Comment (vbar) [inline content/hook-set-codes.json:51]: hmm - why is the above repeated 6 times?

Comment (muzam1l) [reply content/hook-set-codes.json:51]: Goddam copilot 😅

**Review by vbar — APPROVED** [review]

**Pull request merged**